### ISSUE TITLE
sidecar --min-time not honoured when object storage not configured

### ISSUE THREAD

**Issue by milan-holub** [thanos-io/thanos#7950]
**Thanos, Prometheus and Golang version used**:
```
thanos, version 0.34.1 (branch: HEAD, revision: 4cf1559998bf6d8db3f9ca0fde2a00d217d4e23e)
  build user:       root@61db75277a55
  build date:       20240219-17:13:48
  go version:       go1.21.7
  platform:         linux/amd64
  tags:             netgo
```

```
prometheus, version 2.46.0 (branch: HEAD, revision: cbb69e51423565ec40f46e74f4ff2dbb3b7fb4f0)
  build user:       root@42454fc0f41e
  build date:       20230725-12:31:24
  go version:       go1.20.6
  platform:         linux/amd64
  tags:             netgo,builtinassets,stringlabels
```

**Object Storage Provider**:
None

**What happened**:
Running sidecar with --min-time=-3h flag without object storage configured:
```
  - args:
    - sidecar
    - --min-time=-3h
    - --prometheus.url=http://127.0.0.1:9090/
    - --grpc-address=:12901
    - --http-address=:12902
    - --log.level=info
    - --log.format=json
```
Having query configured against this sidecar:
```
   ...
    --endpoint=dnssrv+_grpc-sidecar-2._tcp.observability-thanos-discovery.monitoring.svc
```
(verified the dnssrv+ is correctly resolved to the sidecar)
When sidecar is started it correctly sets the relative minimum time as it can be then seen from the query UI /stores endpoint
<img width="1894" alt="Screenshot 2024-12-02 at 14 31 36" src="https://github.com/user-attachments/assets/26558bb4-72eb-4147-8962-a7c72630f143">
and the queried data are limited to that time.
Unfortunately the minimum time is never updated and as the time passes data can be still queried up to the fixed point in time of <start of the sidecar - 3h> even after days.

**What you expected to happen**:
- I'd expect that the `--min-time` flag is honoured as per documentation:
> Start of time range limit to serve. Thanos sidecar will serve only metrics, which happened later than this value.
- to get only the last 3 hours of data anytime

**How to reproduce it (as minimally and precisely as possible)**:
configure sidecar and query as per above

**Full logs to relevant components**:
No specific log records identified.

**Anything else we need to know**:
Based on [this comment](https://github.com/thanos-io/thanos/blob/main/cmd/thanos/sidecar.go#L125-L126) in the code it seems to be like a forgotten TODO? 

Bumped into this for the use-case of having multiple sidecars per prometheus pod for performance/sizing reasons (separate sidecar for uploads to object storage and another one for grpc query).

**Comment by milan-holub** [comment]
Great summary from dosubot but I think that there are also use-cases out of the object storage upload area. Limiting query results is one of those. Any ideas? Is there a chance that it will be implemented?

**Comment by GiedriusS** [comment]
Is it the same with the newest version?

**Comment by milan-holub** [comment]
just about to confirm it is still the same with the latest release v0.37.1

**Comment by GiedriusS** [comment]
@MichaHoffmann could you take a look?

**Comment by MichaHoffmann** [comment]
I think https://github.com/thanos-io/thanos/blame/7037331e6ea7dbe85a1b7af37bf8ea277a80663d/cmd/thanos/sidecar.go#L508 ~  this should have been a max(configured-mintime, prometheus-mintime) :see-no-evil:

**Comment by MichaHoffmann** [comment]
Fix here: https://github.com/thanos-io/thanos/pull/7970

**Comment by atos-cosmin-gavagiuc** [comment]
> Fix here: #7970

it works, thanks a lot.
